src/charm.py

```diff
@@ -418,6 +416,30 @@ def _report_service_ips(self):
             x.spec.cluster_ip for x in services.items if x.metadata.name in all_k8s_service_names
         ]
 
+    def _report_ingress_ips(self) -> str:
+        """Report on ingress IP(s)."""
+        self.k8s_auth()
+        api = _networking_v1_api()
+
+        # Wait up to `interval * count` seconds for ingress IPs.
+        count, interval = 100, 1
+        for _ in range(count):
+            ingresses = api.list_namespaced_ingress(namespace=self._namespace)
+            # XXX: Will this return multiple IPs if there are any?
```

Comment:
We shouldn't have an XXX comment here - can we confirm this either way and update as appropriate?